.ci/job/vm/vm_runner.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+#
+# Copyright (c) 2021 Red Hat
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Defines the vm_runner interface.
+
+vm_dir=$(dirname "$0")
+cidir="${vm_dir}/../../"
+source "${cidir}/lib.sh"
+
+# The VM engine. Default to Vagrant.
+KATA_TESTS_VM_ENGINE=${KATA_TESTS_VM_ENGINE:-"vagrant"}
+# The default VM name.
+KATA_TESTS_VM_NAME=${KATA_TESTS_VM_NAME:-"fedora"}
+
+# Some jobs cannot run inside a VM. For instance, those which need bare-metal
+# machine. So here it keeps a list of unsupported jobs.
+UNSUPPORTED_JOBS=(
```

Comment:
what about metrics? also I believe vfio needs baremetal

Reply:
No, surprisingly vfio does not run on baremetal right now: it explicitly looks for a second virtio-net device in its "host" to then pass through into the Kata guests.